src/main.js

```diff
@@ -567,17 +567,21 @@ async function resolveNlsConfiguration() {
 		// valid after we have received the app ready event. This is why the
 		// code is here.
 
-		// The ternary and ts-ignore can both be removed once Electron
-		// officially adopts the getPreferredSystemLanguages API.
-		// Ref https://github.com/microsoft/vscode/issues/159813
-		// and https://github.com/electron/electron/pull/36035
 		/**
 		 * @type string
 		 */
-		// @ts-ignore API not yet available in the official Electron
-		let appLocale = ((product.quality === 'insider' || product.quality === 'exploration') && app?.getPreferredSystemLanguages()?.length) ?
-			// @ts-ignore API not yet available in the official Electron
-			app.getPreferredSystemLanguages()[0] : app.getLocale();
+		let appLocale = app.getLocale();
+
+		// This if statement can be simplified once
+		// VS Code moves to Electron 22.
+		// Ref https://github.com/microsoft/vscode/issues/159813
+		// and https://github.com/electron/electron/pull/36035
+		if ((product.quality === 'insider' || product.quality === 'exploration')
+			&& 'getPreferredSystemLanguages' in app
+			&& typeof app.getPreferredSystemLanguages === 'function'
+			&& app.getPreferredSystemLanguages().length) {
+			appLocale = app.getPreferredSystemLanguages()[0];
```

Comment:
Don't you still need `app.getLocale()` for stable?

Reply:
It is set at https://github.com/microsoft/vscode/pull/166884/files#diff-27653c212e1cfe533e4eb2f7d0d3f89604c9de48a09583b4cbbbcbd08a07da79R573